tests/Unit/Domain/Test_DomainCreators.cpp

```diff
@@ -0,0 +1,177 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <algorithm>
+#include <catch.hpp>
+
+#include "DataStructures/IndexIterator.hpp"
+#include "Domain/Block.hpp"
+#include "Domain/DomainHelpers.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+
+// Iterates over the logical corners of a VolumeDim-dimensional cube.
+template <size_t VolumeDim>
+class CornerIterator {
+ public:
+  // This constructor is for iterating over all 2^VolumeDim corners.
+  CornerIterator();
```

Comment:
noexcept everything

Reply:
and we mean EVERYTHING, like literally EVERYTHING (but not the doxygen comments, right Nils?)